### ISSUE TITLE
Error when querying via GraphQL

### ISSUE THREAD

**Issue by ca-miked** [engram-design/SuperTable#300]
### Description
Getting an  error when trying to query a SuperTable field. I have deleted and recreated the field multiple times always resulting in the same error. It looks like GQL support is rather new (#295) so perhaps this is a bug? Thanks for all the efforts on a great plugin btw.

### Steps to reproduce

1. Create a SuperTable field type
2. Add more than one field within the table
3. Query field via GQL

### Additional info

- Plugin version: 2.3.0
- Craft version: 3.3.3

### Error Output

> Schema must contain unique named types but contains multiple types named \"myFieldName_BlockType\". Make sure that `resolveType` function of abstract type \"myFieldName_SuperTableField\" returns the same type instance as referenced anywhere else within the schema (see http://webonyx.github.io/graphql-php/type-system/#type-registry).

**Comment by djurdjen** [comment]
Thanks for the great plugin first of all!
I experience the same issue. Is there perhaps any kind of workaround that can be used in the meantime?

Plugin version: 2.3.0
Craft version: 3.3.3

**Comment by engram-design** [comment]
Can either of you let me know what you're querying with, just so I can check things on my end?

**Comment by CosAnca** [comment]
@engram-design Getting the same error for one of my projects as well.

### Steps to reproduce
1. Create a Matrix field type
2. Create a block that contains multiple fields, one being a SuperTable type field (Blurb in my case)
3. Create multiple various type fields within the SuperTable field (Blurb)
4. On an entry page create one or more Blurbs
5. Querying with the following will throw error

```
{
  entries(section:"work") {
    title
    ...on work_caseStudy_Entry {
      contentBlocks {
        __typename
        ...on contentBlocks_blurbBlock_BlockType {
          sectionIntro
          multicolumn
          blurb {
            ...on blurb_BlockType {
              blurbTitle
            }
          }
        }
      }
    }
  }
}
```

### Error output

>Schema must contain unique named types but contains multiple types named \"blurb_BlockType\". Make sure that `resolveType` function of abstract type \"blurb_SuperTableField\" returns the same type instance as referenced anywhere else within the schema (see http://webonyx.github.io/graphql-php/type-system/#type-registry).

### Additional info

- CraftCMS version: 3.3.4.1
- Plugin version: 2.3.0

Screenshots of the actual block structure, GQL query and output
[Block structure screenshot](https://cl.ly/1672e102ef49)
[GQL query and output screenshot](https://cl.ly/06a2de1ce1e1)

**Comment by sgtpenguin** [comment]
I had this problem too.  Found this in a similar issue with Neo.  It works!

> If you're seeing an error like: Schema must contain unique named types but contains multiple types named..., add the below to the query.
> 
> ```
> __schema {
>     types {
>         name
>         description
>     }
> }
> ```
> For some reason when updating from craft 3.3.2 -> latest, it creates that error. Adding the above, query and then removing fixes it.

https://github.com/spicywebau/craft-neo/issues/252#issuecomment-533959734

**Comment by mattstein** [comment]
@sgtpenguin This works for me when querying directly on Craft's GraphQL endpoint, but `__schema` doesn't seem to be available to query for Gatsby's `gatsby-source-graphql` sources. The same query results in `Cannot query field "__schema" on type "CraftGQL".`

**Comment by ccchapman** [comment]
I am having the same issue as @mattstein.

This is due to [a change in Craft 3.3.3](https://github.com/craftcms/cms/commit/1680ede0252ad135361cf4b352acc84b3024d4bc#diff-261b0349526f623614a6ab4e40af13dfL142-R143).

```diff
-            $devMode = Craft::$app->getConfig()->getGeneral()->devMode;
-            $schemaDef = $gqlService->getSchemaDef($schema, $devMode);
+            $schemaDef = $gqlService->getSchemaDef($schema, StringHelper::contains($query, '__schema'));
```

You can see that Craft previously returned the full schema when the site was in `devMode`. Now, it only returns the full schema when `__schema` is in the query.

I do not know what the proper solution is here. Any thoughts?

cc: @andris-sevcenko @brandonkelly

**Comment by mattstein** [comment]
I can't figure out how to modify `gatsby-source-graphql` to always include `__schema` in its queries or whether this is ultimately a Craft or a Gatsby issue.

What's interesting is that this doesn't seem to be a problem for Matrix blocks, but only for Super Table blocks (and presumably any third-party unions?).

**Comment by dangayle** [comment]
A fix for this is upstream, but needs attention on the plugin side to implement: https://github.com/craftcms/cms/issues/5067

**Comment by wfendler** [comment]
I've read through this issue and the Craft issue a couple times but can't quite tell. Is this something we're waiting on a fix from Craft or does this plugin need updating to fix this?

Gatsby doesn't pull in `__schema` so I'm unable to use the workaround that works on the GraphQL explore page.

**Comment by mattstein** [comment]
@wfendler I believe each plugin author has to implement the fix, including Josh + Super Table, and I'm still patching Craft as of 3.3.11 to get around this.

```diff
diff --git a/src/controllers/GraphqlController.php b/src/controllers/GraphqlController.php
index ad7641af80..4666ed4a71 100644
--- a/src/controllers/GraphqlController.php
+++ b/src/controllers/GraphqlController.php
@@ -140,7 +140,7 @@ class GraphqlController extends Controller
         }
 
         try {
-            $schemaDef = $gqlService->getSchemaDef($schema, StringHelper::contains($query, '__schema'));
+            $schemaDef = $gqlService->getSchemaDef($schema, true);
             $result = $gqlService->executeQuery($schemaDef, $query, $variables, $operationName);
         } catch (\Throwable $e) {
             Craft::$app->getErrorHandler()->logException($e);
```

From what I can tell, [this line](https://github.com/verbb/super-table/blob/craft-3/src/fields/SuperTableField.php#L787) needs to `return $value->getGqlTypeName();`, but I'm not sure if that'd have more implications for the codebase.

**Comment by engram-design** [comment]
Should be fixed in [2.3.1](https://github.com/verbb/super-table/releases/tag/2.3.1)
